packages/frontend/src/components/common/Filters/FilterInputs/DateFilterInputs.tsx

```diff
@@ -266,7 +266,7 @@ const DateFilterInputs = <T extends ConditionalRule = DateFilterRule>(
                         onChange={(
                             range: [Date | null, Date | null] | null,
                         ) => {
-                            if (range && range[0] && range[1]) {
+                            if (range && range[0]) {
```

Comment:
The fix was simple. When we select a new `before` that that happens after `after` date. Bluerprint returns the new range[0] but sets range[1] to undefined, which means that on the previous version, we were never "saving" the new `before` date using `onChange` .

This issue was also happening if you set the `after` before the `first` date. That case is also fixed now.